extensions/amp-auto-ads/0.1/placement.js

```diff
@@ -181,15 +184,44 @@ export class Placement {
         }
         this.adElement_ = this.createAdElement_(baseAttributes);
         this.injector_(this.anchorElement_, this.adElement_);
-        return this.resources_.attemptChangeSize(this.adElement_,
-            TARGET_AD_HEIGHT_PX, undefined, this.margins_)
-            .then(() => {
-              this.state_ = PlacementState.PLACED;
-              return this.state_;
-            }, () => {
-              this.state_ = PlacementState.RESIZE_FAILED;
-              return this.state_;
-            });
+        if (isExperimentOn(this.ampdoc.win,
+            'amp-auto-ads-responsive')) {
+          const viewportWidth = this.resources_.getViewport().getWidth();
+          const viewportHeight = this.resources_.getViewport().getHeight();
+          const responsiveHeight =
+            getResponsiveHeightForContext_(viewportWidth, viewportHeight);
+          Services.resourcesForDoc(this.anchorElement_)
+              .getElementLayoutBox(this.anchorElement_)
+              .then(layoutBox => {
+                const direction =
+                  computedStyle(this.ampdoc.win, this.anchorElement_)
+                      ['direction'];
+                if (direction == 'rtl') {
+                  this.margins_.right = layoutBox.left;
```

Comment:
I don't think this.margins_ should be modified like this. Currently it's passed in via the constructor (and not deep copied), so this could have side effects for any other placements using the same object.

I think it would be better to create a new margin object that merges this.margins_ with the margins required for the responsive logic.

Reply:
Done